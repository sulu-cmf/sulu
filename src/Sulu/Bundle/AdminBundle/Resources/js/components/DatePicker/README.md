The DatePicker component builds on the top of the [react-datetime](https://github.com/YouCanBookMe/react-datetime) library. 

```javascript
initialState = {value: null};

const onChange = (newValue) => {
    setState({value: newValue});
};

<div>
    <div style={{paddingBottom: '50px'}}>Current value: {state.value ? state.value.toLocaleDateString() : 'null'}</div>
    <DatePicker value={state.value} onChange={onChange} />
</div>
```

Default date picker with initial value set:

```javascript
initialState = {value: new Date('2017-05-30')};
const onChange = (newValue) => {
    setState({value: newValue});
};

<div>
    <div style={{paddingBottom: '50px'}}>Current value: {state.value ? state.value.toLocaleDateString() : 'null'}</div>
    <DatePicker value={state.value} onChange={onChange} />
</div>
```

Date time picker can be created with the option property `timeFormat`.

```javascript
initialState = {value: ''};
const onChange = (newValue) => {
    setState({value: newValue});
};

const options = {
    timeFormat: true,
};

<div>
    <div style={{paddingBottom: '50px'}}>Current value: {state.value ? state.value.toLocaleDateString() : 'null'}</div>
    <DatePicker value={state.value} onChange={onChange} options={options} />
</div>
```

Month picker can be created with the option property `dateFormat` set to `MMMM`.

```javascript
initialState = {value: ''};
const onChange = (newValue) => {
    setState({value: newValue});
};

const options = {
    dateFormat: 'MMMM',
};

<div>
    <div style={{paddingBottom: '50px'}}>Current value: {state.value ? state.value.toLocaleDateString() : 'null'}</div>
    <DatePicker value={state.value} onChange={onChange} options={options} />
</div>
```

Year picker can be created with the option property `dateFormat` set to `YYYY`.

```javascript
initialState = {value: ''};
const onChange = (newValue) => {
    setState({value: newValue});
};

const options = {
    dateFormat: 'YYYY',
};

<div>
    <div style={{paddingBottom: '50px'}}>Current value: {state.value ? state.value.toLocaleDateString() : 'null'}</div>
    <DatePicker value={state.value} onChange={onChange} options={options} />
</div>
```

Timepicker is also possible with options `dateFormat` to `false` and `timeFormat` to `true`.

```javascript
initialState = {value: ''};
const onChange = (newValue) => {
    setState({value: newValue});
};

const options = {
    dateFormat: false,
    timeFormat: true,
};

<div>
    <div style={{paddingBottom: '50px'}}>Current value: {state.value ? state.value.toLocaleDateString() : 'null'}</div>
    <DatePicker value={state.value} onChange={onChange} options={options} />
</div>
```

You can tell the component that it's current value isn't valid with the prop `valid`.

Hint: Try to set the date greater than 2018-01-01 ;)

```javascript
initialState = {
    value: '',
    valid: false
};
const onChange = (newValue) => {
    setState({value: newValue, valid: newValue > new Date('2018-01-01')});
};

<div>
    <div style={{paddingBottom: '50px'}}>Current value: {state.value ? state.value.toLocaleDateString() : 'null'}</div>
    <DatePicker value={state.value} valid={state.valid} onChange={onChange} />
</div>
```