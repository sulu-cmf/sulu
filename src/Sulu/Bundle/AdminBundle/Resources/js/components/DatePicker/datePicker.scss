@import '../../containers/Application/colors.scss';

$fontColor: $scorpion;
$backgroundColorActive: $shakespeare;
$borderColorToday: $blueZodiac;
$borderColor: $silver;
$iconColor: $dustyGray;
$iconBorderPadding: 5px;
$placeholderColor: $dustyGray;
$borderWidth: 1px;
$borderRadius: 3px;
$inputHeight: 30px;
$inputColor: $black;
$inputFontSize: 12px;

:global {
    .rdtPicker {
        padding: 0 12px;
        box-shadow: none;
        border: 1px solid $borderColor;
        border-radius: 0 0 $borderRadius $borderRadius;
        color: $fontColor;
        position: inherit;
        width: auto;

        table {
            border-spacing: 0;
        }

        thead td {
            padding-left: 0;
            padding-right: 0;
        }

        thead tr:first-child th {
            border-bottom: 1px solid $borderColor;
            padding: 4px 0 5px;
        }

        td,
        th {
            line-height: 19px;
            height: 19px;
            font-size: 14px;
        }

        th {
            padding: 3px 2px 2px;
        }

        td {
            padding: 8px;

            &:hover {
                border-radius: 3px;
            }

            &.rdtActive,
            &.rdtActive:hover {
                background: $backgroundColorActive;
                border-radius: 3px;
            }

            &.rdtToday::before {
                border-bottom-color: $borderColorToday;
            }
        }
    }

    .rdtDays {
        thead tr:last-child th {
            padding-top: 15px;
        }
    }

    .rdtYears {
        table:last-child {
            padding: 5px;
        }
    }

    .rdtCounters .rdtCount {
        font-size: 14px;
    }
}
