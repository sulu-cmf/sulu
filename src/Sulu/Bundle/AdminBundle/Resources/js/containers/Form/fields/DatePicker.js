// @flow
import React from 'react';
import {computed} from 'mobx';
import {observer} from 'mobx-react';
import moment from 'moment';
import DatePickerComponent from '../../../components/DatePicker';
import type {FieldTypeProps} from '../../../types';

function createStringValue(value: ?Date, format: string) {
    if (!value) {
        return undefined;
    }

    return moment(value).format(format);
}

function getValue(value: ?string, format: string): ?moment {
    if (!value) {
        return undefined;
    }

    const momentObject = moment(value, format);

    if (!momentObject.isValid()) {
        return undefined;
    }

    return momentObject.toDate();
}

@observer
export default class DatePicker extends React.Component<FieldTypeProps<?string>> {
    @computed get format() {
        const {fieldTypeOptions} = this.props;
        const {dateFormat, timeFormat} = fieldTypeOptions;

        if (dateFormat && timeFormat) {
            return 'YYYY-MM-DDTHH:mm:ss';
        }

        if (dateFormat) {
            return 'YYYY-MM-DD';
        }

        return 'HH:mm:ss';
    }

    handleChange = (value: ?Date) => {
        const {onChange, onFinish} = this.props;
        const stringValue = createStringValue(value, this.format);

        onChange(stringValue);
        onFinish();
    };

    render() {
        const {dataPath, disabled, error, fieldTypeOptions, value} = this.props;
        const {dateFormat, timeFormat} = fieldTypeOptions;

        if (dateFormat === undefined || timeFormat === undefined) {
            throw new Error('The "dateFormat" and "timeFormat" fieldTypeOption have to be set!');
        }

        const options = {};

        if (timeFormat) {
            options.timeFormat = timeFormat;
        }

        if (!dateFormat) {
            options.dateFormat = false;
        }

        return (
            <DatePickerComponent
                disabled={!!disabled}
                id={dataPath}
                onChange={this.handleChange}
                options={options}
                valid={!error}
                value={getValue(value, this.format)}
            />
        );
    }
}
