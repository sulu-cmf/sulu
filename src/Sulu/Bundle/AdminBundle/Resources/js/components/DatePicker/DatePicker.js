// @flow
import React from 'react';
import ReactDOM from 'react-dom';
import type {ElementRef} from 'react';
import ReactDatetime from 'react-datetime';
import {observer} from 'mobx-react';
import {action, observable} from 'mobx';
import 'react-datetime/css/react-datetime.css';
import type Moment from 'moment';
import moment from 'moment';
import Input from '../Input';
import Popover from '../Popover';
import './datePicker.scss';

type Props = {
    /** Placeholder can be overwritten */
    placeholder?: string,
    /** Listen for changes of this component */
    onChange: (value: ?Date) => void,
    /** Configure the datepicker to your needs, for more information have a look in the README.md */
    options: Object,
    valid: boolean,
    value: ?Date,
};

@observer
export default class DatePicker extends React.Component<Props> {
    static defaultProps = {
        options: {},
        valid: true,
    };

    inputChanged: boolean = false;

    @observable open: boolean = false;
    @observable showError: boolean = false;
    @observable value: ?string | ?Date | ?Moment = null;
    @observable inputRef: ?ElementRef<*>;

    @action setOpen(open: boolean) {
        this.open = open;
    }

    @action setValue(value: ?string | ?Date | ?Moment) {
        this.value = value;
    }

    @action setShowError(showError: boolean) {
        this.showError = showError;
    }

    @action setInputRef = (ref: ?ElementRef<*>) => {
        this.inputRef = ref;
    };

    componentWillMount() {
        this.setValue(this.props.value);
    }

    componentWillReceiveProps(nextProps: Props) {
        if (this.value && !nextProps.value) {
            return;
        }

        this.setValue(nextProps.value);
    }

    handleChange = (date: ?Date) => {
        this.inputChanged = false;
        this.props.onChange(date);

        this.setShowError(!!this.value && !date);
    };

    handleDatepickerChange = (date: string | Moment) => {
        if (!date) {
            this.setValue(undefined);
            this.handleChange(undefined);

            return;
        }

        if (typeof date === 'string') {
            this.setValue(date);

            return;
        }

        if (!date.isValid()) {
            this.handleChange(undefined);

            return;
        }

        this.handleChange(date.toDate());
    };

    handleInputBlur = () => {
        if (this.inputChanged && typeof this.value === 'string') {
            this.handleChange(undefined);
        }
    };

    handleOpenOverlay = () => {
        this.setOpen(true);
    };

    handleCloseOverlay = () => {
        this.setOpen(false);
    };

    getInputChange = (props: Object) => {
        return (value: ?string, event: SyntheticEvent<HTMLInputElement>) => {
            this.inputChanged = true;
            props.onChange(event);
        };
    };

    getPlaceholder = (fieldOptions: {} => string) => {
        const placeholderDate = fieldOptions.dateFormat ? fieldOptions.dateFormat : '';
        const placeholderTime = fieldOptions.timeFormat ? moment.localeData().longDateFormat('LT') : '';

        if (!fieldOptions.timeFormat) {
            return placeholderDate;
        } else if (fieldOptions.dateFormat && fieldOptions.timeFormat) {
            return placeholderDate + ' ' + placeholderTime;
        }

        return placeholderTime;
    };

    renderInput = (props: Object) => {
        const handleInputChange = this.getInputChange(props);

        if (!this.inputRef) {
            return null;
        }

        return ReactDOM.createPortal(
            <Input
                {...props}
                onIconClick={this.handleOpenOverlay}
                onChange={handleInputChange}
                onBlur={this.handleInputBlur}
            />,
            this.inputRef
        );
    };

    render() {
        const {options, placeholder, valid} = this.props;

        const fieldOptions = {
            closeOnSelect: true,
            timeFormat: false,
            dateFormat: moment.localeData().longDateFormat('L'),
            ...options,
        };

        const inputProps = {
            placeholder: placeholder ? placeholder : this.getPlaceholder(fieldOptions),
            valid: valid && !this.showError,
            icon: fieldOptions.dateFormat ? 'su-calendar' : 'su-clock',
        };

        return (
            <div>
                <div ref={this.setInputRef} />
                <Popover
                    open={true}
                    anchorElement={this.inputRef}
                    backdrop={this.open}
                    onClose={this.handleCloseOverlay}
                    verticalOffset={-31}
                    horizontalOffset={34}
                >
                    {
                        (setPopoverRef, styles) => (
                            <div ref={setPopoverRef} style={styles}>
                                <ReactDatetime
                                    {...fieldOptions}
                                    inputProps={inputProps}
                                    renderInput={this.renderInput}
                                    open={this.open}
                                    value={this.value}
                                    onChange={this.handleDatepickerChange}
                                    onBlur={this.handleCloseOverlay}
                                />
                            </div>
                        )
                    }
                </Popover>
            </div>
        );
    }
}
