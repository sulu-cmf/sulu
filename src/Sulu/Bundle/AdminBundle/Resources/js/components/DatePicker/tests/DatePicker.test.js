// @flow
import React from 'react';
import {mount} from 'enzyme';
import moment from 'moment-timezone';
import pretty from 'pretty';
import DatePicker from '../DatePicker';

beforeEach(() => {
    const constantDate = new Date(Date.UTC(2017, 3, 15, 6, 32, 20));
    (Date: any).now = jest.fn().mockReturnValue(constantDate);

    moment.tz.setDefault('Europe/Vienna');
});

test('DatePicker should render', () => {
    const onChange = jest.fn();
    const datePicker = mount(<DatePicker onChange={onChange} value={null} />);

    expect(datePicker.render()).toMatchSnapshot();
    expect(pretty(datePicker.find('DateTime').html())).toMatchSnapshot();
});

test('DatePicker should render when disabled', () => {
    const onChange = jest.fn();
    const value = new Date('2017-05-23');
    const datePicker = mount(<DatePicker disabled={true} onChange={onChange} value={value} />);

    expect(datePicker.render()).toMatchSnapshot();
});

test('DatePicker should open overlay on icon-click', () => {
    const onChange = jest.fn();
    const datePicker = mount(<DatePicker onChange={onChange} value={null} />);

    expect(datePicker.find('DateTime').props().open).toBeFalsy();
    datePicker.find('Icon').simulate('click');
    expect(datePicker.find('DateTime').props().open).toBeTruthy();
});

test('DatePicker should not open overlay on icon-click when disabled', () => {
    const onChange = jest.fn();
    const datePicker = mount(<DatePicker disabled={true} onChange={onChange} value={null} />);

    expect(datePicker.find('DateTime').props().open).toBeFalsy();
    datePicker.find('Icon').simulate('click');
    expect(datePicker.find('DateTime').props().open).toBeFalsy();
});

test('DatePicker should render with placeholder', () => {
    const onChange = jest.fn();
    const datePicker = mount(<DatePicker onChange={onChange} placeholder="My placeholder" value={null} />);

    expect(datePicker.render()).toMatchSnapshot();
});

test('DatePicker should render with value', () => {
    const onChange = jest.fn();
    const value = new Date('2017-05-23');
    const datePicker = mount(<DatePicker onChange={onChange} value={value} />);

    expect(datePicker.render()).toMatchSnapshot();
});

test('DatePicker should render null value as empty string', () => {
    const onChange = jest.fn();
    const datePicker = mount(<DatePicker onChange={onChange} value={null} />);

    expect(datePicker.render()).toMatchSnapshot();
});

test('DatePicker should render date format only with month', () => {
    const onChange = jest.fn();
    const options = {
        dateFormat: 'MMMM',
    };
    const datePicker = mount(<DatePicker onChange={onChange} options={options} value={null} />);

    expect(datePicker.render()).toMatchSnapshot();
});

test('DatePicker should render date format only with year', () => {
    const onChange = jest.fn();
    const options = {
        dateFormat: 'YYYY',
    };
    const datePicker = mount(<DatePicker onChange={onChange} options={options} value={null} />);

    expect(datePicker.render()).toMatchSnapshot();
});

test('DatePicker should render date picker with time picker', () => {
    const onChange = jest.fn();
    const options = {
        timeFormat: true,
    };
    const datePicker = mount(<DatePicker onChange={onChange} options={options} value={null} />);

    expect(datePicker.render()).toMatchSnapshot();
    expect(pretty(datePicker.find('DateTime').html())).toMatchSnapshot();
});

test('DatePicker should render error', () => {
    const onChange = jest.fn();
    const datePicker = mount(<DatePicker onChange={onChange} valid={false} value={null} />);

    expect(datePicker.render()).toMatchSnapshot();
});

test('DatePicker should render error when invalid value is set', () => {
    const onChange = jest.fn();
    const options = {
        dateFormat: 'YYYY',
    };
    const datePicker = mount(<DatePicker onChange={onChange} options={options} value={null} />);

    // check if showError is set correctly
    datePicker.find('Input').instance().props.onChange('xxx', {target: {value: 'xxx'}});
    datePicker.find('Input').instance().props.onBlur();
    datePicker.update();
    expect(datePicker.instance().showError).toBe(true);

    // snapshot
    expect(datePicker.render()).toMatchSnapshot();

    // now add a valid value
    datePicker.find('Input').instance().props.onChange('2018', {target: {value: '2018'}});
    datePicker.find('Input').instance().props.onBlur();
    datePicker.update();
    expect(datePicker.instance().showError).toBe(false);
});

test('DatePicker should set class correctly when overlay was opened/closed', () => {
    const onChange = jest.fn();
    const input = mount(<DatePicker onChange={onChange} value={null} />);

    // overlay should be closed
    expect(input.find('div.rdt').hasClass('rdtOpen')).toBe(false);

    // open dialog and check if class is set
    input.find('Input Icon span').simulate('click');
    expect(input.find('div.rdt').hasClass('rdtOpen')).toBe(true);

    // choose a date and check if class was removed again
    input.find('.rdtPicker tbody tr td').first().simulate('click');
    expect(input.find('div.rdt').hasClass('rdtOpen')).toBe(false);

    // check if value is in input
    expect(input.find('Input').prop('value')).toBe('03/26/2017');
});
