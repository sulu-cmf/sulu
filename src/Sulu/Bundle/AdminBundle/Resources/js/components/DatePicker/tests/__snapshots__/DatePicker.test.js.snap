// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker should render 1`] = `
<div>
  <div>
    <label
      class="input default left"
    >
      <div
        class="prependedContainer default"
      >
        <span
          aria-label="su-calendar"
          class="su-calendar clickable icon default iconClickable"
          role="button"
          tabindex="0"
        />
      </div>
      <input
        placeholder="MM/DD/YYYY"
        type="text"
        value=""
      />
    </label>
  </div>
</div>
`;

exports[`DatePicker should render 2`] = `
"<div class=\\"rdt\\">
  <div></div>
  <div class=\\"rdtPicker\\">
    <div class=\\"rdtDays\\">
      <table>
        <thead>
          <tr>
            <th class=\\"rdtPrev\\"><span>‹</span></th>
            <th class=\\"rdtSwitch\\" colspan=\\"5\\" data-value=\\"3\\">April 2017</th>
            <th class=\\"rdtNext\\"><span>›</span></th>
          </tr>
          <tr>
            <th class=\\"dow\\">Su</th>
            <th class=\\"dow\\">Mo</th>
            <th class=\\"dow\\">Tu</th>
            <th class=\\"dow\\">We</th>
            <th class=\\"dow\\">Th</th>
            <th class=\\"dow\\">Fr</th>
            <th class=\\"dow\\">Sa</th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td data-value=\\"26\\" class=\\"rdtDay rdtOld\\">26</td>
            <td data-value=\\"27\\" class=\\"rdtDay rdtOld\\">27</td>
            <td data-value=\\"28\\" class=\\"rdtDay rdtOld\\">28</td>
            <td data-value=\\"29\\" class=\\"rdtDay rdtOld\\">29</td>
            <td data-value=\\"30\\" class=\\"rdtDay rdtOld\\">30</td>
            <td data-value=\\"31\\" class=\\"rdtDay rdtOld\\">31</td>
            <td data-value=\\"1\\" class=\\"rdtDay\\">1</td>
          </tr>
          <tr>
            <td data-value=\\"2\\" class=\\"rdtDay\\">2</td>
            <td data-value=\\"3\\" class=\\"rdtDay\\">3</td>
            <td data-value=\\"4\\" class=\\"rdtDay\\">4</td>
            <td data-value=\\"5\\" class=\\"rdtDay\\">5</td>
            <td data-value=\\"6\\" class=\\"rdtDay\\">6</td>
            <td data-value=\\"7\\" class=\\"rdtDay\\">7</td>
            <td data-value=\\"8\\" class=\\"rdtDay\\">8</td>
          </tr>
          <tr>
            <td data-value=\\"9\\" class=\\"rdtDay\\">9</td>
            <td data-value=\\"10\\" class=\\"rdtDay\\">10</td>
            <td data-value=\\"11\\" class=\\"rdtDay\\">11</td>
            <td data-value=\\"12\\" class=\\"rdtDay\\">12</td>
            <td data-value=\\"13\\" class=\\"rdtDay\\">13</td>
            <td data-value=\\"14\\" class=\\"rdtDay\\">14</td>
            <td data-value=\\"15\\" class=\\"rdtDay rdtToday\\">15</td>
          </tr>
          <tr>
            <td data-value=\\"16\\" class=\\"rdtDay\\">16</td>
            <td data-value=\\"17\\" class=\\"rdtDay\\">17</td>
            <td data-value=\\"18\\" class=\\"rdtDay\\">18</td>
            <td data-value=\\"19\\" class=\\"rdtDay\\">19</td>
            <td data-value=\\"20\\" class=\\"rdtDay\\">20</td>
            <td data-value=\\"21\\" class=\\"rdtDay\\">21</td>
            <td data-value=\\"22\\" class=\\"rdtDay\\">22</td>
          </tr>
          <tr>
            <td data-value=\\"23\\" class=\\"rdtDay\\">23</td>
            <td data-value=\\"24\\" class=\\"rdtDay\\">24</td>
            <td data-value=\\"25\\" class=\\"rdtDay\\">25</td>
            <td data-value=\\"26\\" class=\\"rdtDay\\">26</td>
            <td data-value=\\"27\\" class=\\"rdtDay\\">27</td>
            <td data-value=\\"28\\" class=\\"rdtDay\\">28</td>
            <td data-value=\\"29\\" class=\\"rdtDay\\">29</td>
          </tr>
          <tr>
            <td data-value=\\"30\\" class=\\"rdtDay\\">30</td>
            <td data-value=\\"1\\" class=\\"rdtDay rdtNew\\">1</td>
            <td data-value=\\"2\\" class=\\"rdtDay rdtNew\\">2</td>
            <td data-value=\\"3\\" class=\\"rdtDay rdtNew\\">3</td>
            <td data-value=\\"4\\" class=\\"rdtDay rdtNew\\">4</td>
            <td data-value=\\"5\\" class=\\"rdtDay rdtNew\\">5</td>
            <td data-value=\\"6\\" class=\\"rdtDay rdtNew\\">6</td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>"
`;

exports[`DatePicker should render date format only with month 1`] = `
<div>
  <div>
    <label
      class="input default left"
    >
      <div
        class="prependedContainer default"
      >
        <span
          aria-label="su-calendar"
          class="su-calendar clickable icon default iconClickable"
          role="button"
          tabindex="0"
        />
      </div>
      <input
        placeholder="MMMM"
        type="text"
        value=""
      />
    </label>
  </div>
</div>
`;

exports[`DatePicker should render date format only with year 1`] = `
<div>
  <div>
    <label
      class="input default left"
    >
      <div
        class="prependedContainer default"
      >
        <span
          aria-label="su-calendar"
          class="su-calendar clickable icon default iconClickable"
          role="button"
          tabindex="0"
        />
      </div>
      <input
        placeholder="YYYY"
        type="text"
        value=""
      />
    </label>
  </div>
</div>
`;

exports[`DatePicker should render date picker with time picker 1`] = `
<div>
  <div>
    <label
      class="input default left"
    >
      <div
        class="prependedContainer default"
      >
        <span
          aria-label="su-calendar"
          class="su-calendar clickable icon default iconClickable"
          role="button"
          tabindex="0"
        />
      </div>
      <input
        placeholder="MM/DD/YYYY h:mm A"
        type="text"
        value=""
      />
    </label>
  </div>
</div>
`;

exports[`DatePicker should render date picker with time picker 2`] = `
"<div class=\\"rdt\\">
  <div></div>
  <div class=\\"rdtPicker\\">
    <div class=\\"rdtDays\\">
      <table>
        <thead>
          <tr>
            <th class=\\"rdtPrev\\"><span>‹</span></th>
            <th class=\\"rdtSwitch\\" colspan=\\"5\\" data-value=\\"3\\">April 2017</th>
            <th class=\\"rdtNext\\"><span>›</span></th>
          </tr>
          <tr>
            <th class=\\"dow\\">Su</th>
            <th class=\\"dow\\">Mo</th>
            <th class=\\"dow\\">Tu</th>
            <th class=\\"dow\\">We</th>
            <th class=\\"dow\\">Th</th>
            <th class=\\"dow\\">Fr</th>
            <th class=\\"dow\\">Sa</th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td data-value=\\"26\\" class=\\"rdtDay rdtOld\\">26</td>
            <td data-value=\\"27\\" class=\\"rdtDay rdtOld\\">27</td>
            <td data-value=\\"28\\" class=\\"rdtDay rdtOld\\">28</td>
            <td data-value=\\"29\\" class=\\"rdtDay rdtOld\\">29</td>
            <td data-value=\\"30\\" class=\\"rdtDay rdtOld\\">30</td>
            <td data-value=\\"31\\" class=\\"rdtDay rdtOld\\">31</td>
            <td data-value=\\"1\\" class=\\"rdtDay\\">1</td>
          </tr>
          <tr>
            <td data-value=\\"2\\" class=\\"rdtDay\\">2</td>
            <td data-value=\\"3\\" class=\\"rdtDay\\">3</td>
            <td data-value=\\"4\\" class=\\"rdtDay\\">4</td>
            <td data-value=\\"5\\" class=\\"rdtDay\\">5</td>
            <td data-value=\\"6\\" class=\\"rdtDay\\">6</td>
            <td data-value=\\"7\\" class=\\"rdtDay\\">7</td>
            <td data-value=\\"8\\" class=\\"rdtDay\\">8</td>
          </tr>
          <tr>
            <td data-value=\\"9\\" class=\\"rdtDay\\">9</td>
            <td data-value=\\"10\\" class=\\"rdtDay\\">10</td>
            <td data-value=\\"11\\" class=\\"rdtDay\\">11</td>
            <td data-value=\\"12\\" class=\\"rdtDay\\">12</td>
            <td data-value=\\"13\\" class=\\"rdtDay\\">13</td>
            <td data-value=\\"14\\" class=\\"rdtDay\\">14</td>
            <td data-value=\\"15\\" class=\\"rdtDay rdtToday\\">15</td>
          </tr>
          <tr>
            <td data-value=\\"16\\" class=\\"rdtDay\\">16</td>
            <td data-value=\\"17\\" class=\\"rdtDay\\">17</td>
            <td data-value=\\"18\\" class=\\"rdtDay\\">18</td>
            <td data-value=\\"19\\" class=\\"rdtDay\\">19</td>
            <td data-value=\\"20\\" class=\\"rdtDay\\">20</td>
            <td data-value=\\"21\\" class=\\"rdtDay\\">21</td>
            <td data-value=\\"22\\" class=\\"rdtDay\\">22</td>
          </tr>
          <tr>
            <td data-value=\\"23\\" class=\\"rdtDay\\">23</td>
            <td data-value=\\"24\\" class=\\"rdtDay\\">24</td>
            <td data-value=\\"25\\" class=\\"rdtDay\\">25</td>
            <td data-value=\\"26\\" class=\\"rdtDay\\">26</td>
            <td data-value=\\"27\\" class=\\"rdtDay\\">27</td>
            <td data-value=\\"28\\" class=\\"rdtDay\\">28</td>
            <td data-value=\\"29\\" class=\\"rdtDay\\">29</td>
          </tr>
          <tr>
            <td data-value=\\"30\\" class=\\"rdtDay\\">30</td>
            <td data-value=\\"1\\" class=\\"rdtDay rdtNew\\">1</td>
            <td data-value=\\"2\\" class=\\"rdtDay rdtNew\\">2</td>
            <td data-value=\\"3\\" class=\\"rdtDay rdtNew\\">3</td>
            <td data-value=\\"4\\" class=\\"rdtDay rdtNew\\">4</td>
            <td data-value=\\"5\\" class=\\"rdtDay rdtNew\\">5</td>
            <td data-value=\\"6\\" class=\\"rdtDay rdtNew\\">6</td>
          </tr>
        </tbody>
        <tfoot>
          <tr>
            <td colspan=\\"7\\" class=\\"rdtTimeToggle\\">12:00 AM</td>
          </tr>
        </tfoot>
      </table>
    </div>
  </div>
</div>"
`;

exports[`DatePicker should render error 1`] = `
<div>
  <div>
    <label
      class="input default left error"
    >
      <div
        class="prependedContainer default"
      >
        <span
          aria-label="su-calendar"
          class="su-calendar clickable icon default iconClickable"
          role="button"
          tabindex="0"
        />
      </div>
      <input
        placeholder="MM/DD/YYYY"
        type="text"
        value=""
      />
    </label>
  </div>
</div>
`;

exports[`DatePicker should render error when invalid value is set 1`] = `
<div>
  <div>
    <label
      class="input default left error"
    >
      <div
        class="prependedContainer default"
      >
        <span
          aria-label="su-calendar"
          class="su-calendar clickable icon default iconClickable"
          role="button"
          tabindex="0"
        />
      </div>
      <input
        placeholder="YYYY"
        type="text"
        value="xxx"
      />
    </label>
  </div>
</div>
`;

exports[`DatePicker should render null value as empty string 1`] = `
<div>
  <div>
    <label
      class="input default left"
    >
      <div
        class="prependedContainer default"
      >
        <span
          aria-label="su-calendar"
          class="su-calendar clickable icon default iconClickable"
          role="button"
          tabindex="0"
        />
      </div>
      <input
        placeholder="MM/DD/YYYY"
        type="text"
        value=""
      />
    </label>
  </div>
</div>
`;

exports[`DatePicker should render when disabled 1`] = `
<div>
  <div>
    <label
      class="input default left disabled"
    >
      <div
        class="prependedContainer default"
      >
        <span
          aria-label="su-calendar"
          class="su-calendar icon default"
        />
      </div>
      <input
        disabled=""
        placeholder="MM/DD/YYYY"
        type="text"
        value="05/23/2017"
      />
    </label>
  </div>
</div>
`;

exports[`DatePicker should render with placeholder 1`] = `
<div>
  <div>
    <label
      class="input default left"
    >
      <div
        class="prependedContainer default"
      >
        <span
          aria-label="su-calendar"
          class="su-calendar clickable icon default iconClickable"
          role="button"
          tabindex="0"
        />
      </div>
      <input
        placeholder="My placeholder"
        type="text"
        value=""
      />
    </label>
  </div>
</div>
`;

exports[`DatePicker should render with value 1`] = `
<div>
  <div>
    <label
      class="input default left"
    >
      <div
        class="prependedContainer default"
      >
        <span
          aria-label="su-calendar"
          class="su-calendar clickable icon default iconClickable"
          role="button"
          tabindex="0"
        />
      </div>
      <input
        placeholder="MM/DD/YYYY"
        type="text"
        value="05/23/2017"
      />
    </label>
  </div>
</div>
`;
